components/tabs/src/sync/record.rs

```diff
@@ -27,7 +28,163 @@ pub struct TabsRecord {
 impl TabsRecord {
     #[inline]
     pub fn from_payload(payload: sync15::Payload) -> Result<Self> {
-        let record: TabsRecord = payload.into_record()?;
-        Ok(record)
+        let client_name: String = payload.data["clientName"]
```

Comment:
Please leave a brief inline comment here about why we have to parse this by hand rather than letting serde do its magic.